sql/planbuilder/from.go

```diff
@@ -57,15 +57,6 @@ func (b *Builder) buildFrom(inScope *scope, te ast.TableExprs) (outScope *scope)
 	return b.buildDataSource(inScope, te[0])
 }
 
-func (b *Builder) validateJoinTableNames(leftScope, rightScope *scope) {
```

Comment:
would be nice to keep this, is the snowball of changes making `scope.tables :[]string->[]sql.TableId` difficult in surprising ways?

Reply:
I agree. I think we want this. 

That said, while I don't know that making this change would be difficult, I also don't know that it will be easy. The PR is already quite large and touches a lot of things: if it's at a point where it's correct and valuable, I think we should merge it handle stuff like this in a follow up. I can add a TODO here though.

Specifically, I think the final goal is for all of the new logic in `getTableAliases` to happen during plan building instead, which would do the same validation we were doing here.